providers/flipt/src/main/java/dev/openfeature/contrib/providers/flipt/FliptProvider.java

```diff
@@ -0,0 +1,240 @@
+            .errorCode(valueProviderEvaluation.getErrorCode())
+            .reason(valueProviderEvaluation.getReason())
+            .flagMetadata(valueProviderEvaluation.getFlagMetadata())
+            .build();
+    }
+
+    private static Integer getIntegerValue(ProviderEvaluation<Value> valueProviderEvaluation, Integer defaultValue) {
+        String valueStr = valueProviderEvaluation.getValue().asObject().toString();
+        try {
+            return Integer.parseInt(valueStr);
+        } catch (NumberFormatException ex) {
+            return defaultValue;
+        }
+    }
+
+    @Override
+    public ProviderEvaluation<Double> getDoubleEvaluation(String key, Double defaultValue, EvaluationContext ctx) {
+        ProviderEvaluation<Value> valueProviderEvaluation = getObjectEvaluation(key, new Value(defaultValue), ctx);
+        Double value = getDoubleValue(valueProviderEvaluation, defaultValue);
+        return ProviderEvaluation.<Double>builder()
+            .value(value)
+            .variant(valueProviderEvaluation.getVariant())
+            .errorCode(valueProviderEvaluation.getErrorCode())
+            .reason(valueProviderEvaluation.getReason())
+            .flagMetadata(valueProviderEvaluation.getFlagMetadata())
+            .build();
+    }
+
+    private static Double getDoubleValue(ProviderEvaluation<Value> valueProviderEvaluation, Double defaultValue) {
+        String valueStr = valueProviderEvaluation.getValue().asObject().toString();
+        try {
+            return Double.parseDouble(valueStr);
+        } catch (NumberFormatException ex) {
+            return defaultValue;
+        }
+    }
+
+    @Override
+    public ProviderEvaluation<Value> getObjectEvaluation(String key, Value defaultValue, EvaluationContext ctx) {
+        if (!ProviderState.READY.equals(state)) {
+            if (ProviderState.NOT_READY.equals(state)) {
+                throw new ProviderNotReadyError(PROVIDER_NOT_YET_INITIALIZED);
+            }
+            throw new GeneralError(UNKNOWN_ERROR);
+        }
+        Map<String, String> contextMap = buildContextMap(ctx);
+        EvaluationRequest request = EvaluationRequest.builder().namespaceKey(fliptProviderConfig.getNamespace())
+            .flagKey(key).entityId(ctx.getTargetingKey()).context(contextMap).build();
+
+        VariantEvaluationResponse response;
+        try {
+            response = fliptApiClient.evaluation().variant(request);
+        } catch (Exception e) {
+            log.error("Error evaluating variant", e);
+            return ProviderEvaluation.<Value>builder()
+                .value(defaultValue)
+                .errorMessage(e.getMessage())
+                .build();
+        }
+
+        if (!response.getMatch()) {
+            log.debug("non matching variant for {}", key);
+            return ProviderEvaluation.<Value>builder()
+                .value(defaultValue)
+                .reason(DEFAULT.name())
+                .build();
```

Comment:
I'm wonder if this should actually be an error? If there's no match for the variant?

Reply:
I don't think it should be an error.
Test case:
- flag configured with variant=x for segment by userId==1.
- flag is evaluated without userId.

In this case, there is no match, API evaluates to no match.
Provider returning the default value.
I am adjusting the reason though to reflect it better.

Reply:
Ah, I see.